apps/web/components/dashboard/tags/AllTagsView.tsx

```diff
@@ -104,22 +187,73 @@ export default function AllTagsView({
   const tagsToPill = (tags: typeof allTags) => {
     let tagPill;
     if (tags.length) {
-      tagPill = tags.map((t) => (
-        <TagPill key={t.id} id={t.id} name={t.name} count={t.count} />
-      ));
+      tagPill = (
+        <div className="flex flex-wrap gap-3">
+          {tags.map((t) => (
+            <Draggable
+              key={t.id}
+              axis="both"
+              onStart={handleDragStart}
+              onDrag={handleDrag}
+              onStop={handleDragEnd}
+              disabled={!draggingEnabled}
+              defaultClassNameDragging={
+                "position-relative z-10 pointer-events-none"
+              }
+              position={
+                !currentState.dragId
+                  ? { x: currentState.x ?? 0, y: currentState.y ?? 0 }
+                  : undefined
+              }
+            >
+              <div
+                className="group relative flex cursor-grab"
+                data-id={t.id}
+                onDragOver={handleDrag}
+              >
+                <TagPill
+                  key={t.id}
+                  id={t.id}
+                  name={t.name}
+                  count={t.count}
+                  isDraggable={draggingEnabled}
+                />
+              </div>
+            </Draggable>
+          ))}
+        </div>
+      );
     } else {
       tagPill = "No Tags";
     }
     return tagPill;
   };
   return (
     <>
+      <label>
+        <input
+          type="checkbox"
+          checked={draggingEnabled}
+          onChange={handleDraggableChange}
+        />
+        Allow Merging via Drag&Drop
+      </label>
+      <br />
+      <label>
+        <input
+          type="checkbox"
+          checked={sortByName}
+          onChange={handleSortByNameChange}
+        />
+        Sort by Name
+      </label>
```

Comment:
I think those can probably be `shadcn` toggles (https://ui.shadcn.com/docs/components/toggle) and can probably be aligned right in the page.
If UI design is not your expertise, I'm happy to implement it myself after merging your PR.